packages/dashboard/src/components/DashboardFilters/DashboardFiltersEditor/components/FilterEditor.tsx

```diff
@@ -131,11 +134,28 @@ const FilterEditor: React.FC<FilterEditorProps> = ({
     loadColumnData()
   }, [config.datasets, config.dashboard.sharedFilters])
 
-  const updateAPIFilter = (key: keyof APIFilter, value: string | boolean) => {
-    const filterClone = _.cloneDeep(filter)
-    const _filter = filterClone.apiFilter || { apiEndpoint: '', valueSelector: '', textSelector: '' }
-    const newAPIFilter: APIFilter = { ..._filter, [key]: value }
+  const updateAPIFilter = (
+    APIEndpoint,
+    APIValueSelector,
+    APITextSelector,
+    APISubgroupValueSelector,
+    APISubgroupTextSelector
+  ) => {
+    const newAPIFilter = !isNestedDropdown
+      ? {
+          apiEndpoint: APIEndpoint,

```

Comment:
if the param wasn't capitalized you could just do 

{
apiEndpoint, 

and assign and name at the same time...